logging/google/cloud/logging/handlers/app_engine.py

```diff
@@ -27,47 +27,63 @@
+
+class AppEngineHandler(CloudLoggingHandler):
+    """A handler that directly makes Stackdriver logging API calls.
+
+    This handler can be used to route Python standard logging messages directly
+    to the Stackdriver Logging API.
+
+    This handler supports both an asynchronous and synchronous transport.
+
+    :type client: :class:`google.cloud.logging.client`
+    :param client: the authenticated Google Cloud Logging client for this
+                   handler to use
+
+    :type name: str
+    :param name: the name of the custom log in Stackdriver Logging. Defaults
+                 to 'python'. The name of the Python logger will be represented
+                 in the ``python_logger`` field.
+
+    :type transport: type
+    :param transport: Class for creating new transport objects. It should
+                      extend from the base :class:`.Transport` type and
+                      implement :meth`.Transport.send`. Defaults to
+                      :class:`.BackgroundThreadTransport`. The other
+                      option is :class:`.SyncTransport`.
+
+    :type resource: :class:`~google.cloud.logging.resource.Resource`
+    :param resource: Monitored resource of the entry, defaults
+                     to the global resource type.
     """
 
-    def __init__(self):
-        """Construct the handler
+    def __init__(self, client,
+                 name=DEFAULT_LOGGER_NAME,
+                 transport=BackgroundThreadTransport,
+                 resource=_GLOBAL_RESOURCE):
+        super(AppEngineHandler, self).__init__(client, name, transport)
+        self.resource=resource
 
-        Large log entries will get mangled if multiple workers write to the
-        same file simultaneously, so we'll use the worker's PID to pick a log
-        filename.
-        """
-        self.filename = _LOG_PATH_TEMPLATE.format(pid=os.getpid())
-        super(AppEngineHandler, self).__init__(self.filename,
-                                               maxBytes=_MAX_LOG_BYTES,
-                                               backupCount=_LOG_FILE_COUNT)
+    def emit(self, record):
+        """Actually log the specified logging record.
 
-    def format(self, record):
-        """Format the specified record into the expected JSON structure.
+        Overrides the default emit behavior of ``StreamHandler``.
 
-        :type record: :class:`~logging.LogRecord`
-        :param record: the log record
+        See: https://docs.python.org/2/library/logging.html#handler-objects
 
-        :rtype: str
-        :returns: JSON str to be written to the log file
+        :type record: :class:`logging.LogRecord`
+        :param record: The record to be logged.
         """
-        message = super(AppEngineHandler, self).format(record)
-        return format_stackdriver_json(record, message)
+        message = super(CloudLoggingHandler, self).format(record)
+        self.transport.send(record, message, self.resource)
```

Comment:
I'd actually prefer you to add the `resource` property to the superclass `CloudLoggingHandler` and therefore you won't need to override this method.